packages/eslint-plugin/tests/rules/no-floating-promises.test.ts

```diff
@@ -593,6 +593,30 @@ doSomething();
         },
       ],
     },
+    {
+      code: `
+declare const myTag: (strings: TemplateStringsArray) => Promise<void>;
+
+myTag\`abc\`;
+myTag\`abc\`.then(() => {});
+myTag\`abc\`.catch(() => {});
+myTag\`abc\`.finally(() => {});
```

Comment:
[Testing] Per https://typescript-eslint.io/contributing/local-development#granular-unit-tests, we generally ask that tests are granular and single-purpose. Could you split this up into one test per error please?

Reply:
Sure, I can separate cases but I just thought this is granular enough. For example, existing test case is for function call is [this](https://github.com/naruaway/typescript-eslint/blob/d5f77d4dd299cac41cb50f3952a0a06b1f8262ea/packages/eslint-plugin/tests/rules/no-floating-promises.test.ts#L509-L537):
```typescript
async function test() {
  Promise.resolve('value');
  Promise.resolve('value').then(() => {});
  Promise.resolve('value').catch();
  Promise.resolve('value').finally();
}
```

Reply:
Now I separated the test cases in 7c8c9dda34b16e98ecfb0adf2561fc4defed6a48. Not bad as I thought 👍

Reply:
Yeah a lot of not-new rules have tests that aren't quite the same as our standards 😅. Thanks!